libs/ktem/ktem/tags/crud.py

```diff
@@ -0,0 +1,244 @@
+            ), "Invalid valid classes!"
+            meta["valid_classes"] = valid_classes
+
+        with Session(self._engine) as session:
+            # tag name must be unique
+            existed_result = self.query_by_name(name)
+            if existed_result:
+                raise Exception(f"Tag name: {name} has already been existed!")
+
+            tag: Tag = Tag(
+                name=name,
+                prompt=prompt,
+                config=config,
+                type=type,
+                scope=scope,
+                meta=meta,
+            )
+            session.add(tag)
+            session.commit()
+
+            return tag.id
+
+    def query_by_id(self, tag_id: str) -> BaseTag | None:
+        with Session(self._engine) as session:
+            statement = select(Tag).where(Tag.id == tag_id)
+
+            result = session.exec(statement).first()
+            return result
+
+    def query_by_ids(self, tag_ids: list[str]) -> list[dict]:
+        with Session(self._engine) as session:
+            statement = select(Tag).where(Tag.id.in_(tag_ids))
+
+            results = session.exec(statement).all()
+            results = [dict(result) for result in results]
+        return results
+
+    def query_by_name(self, tag_name: str) -> BaseTag | None:
+        with Session(self._engine) as session:
+            statement = select(Tag).where(Tag.name == tag_name)
+
+            result = session.exec(statement).first()
+            return result
+
+    def delete_by_name(self, tag_name: str) -> bool:
+        with Session(self._engine) as session:
+            statement = select(Tag).where(Tag.name == tag_name)
+            result = session.exec(statement).first()
+
+            if result:
+                session.delete(result)
+                session.commit()
+                return True
+
+        return False
+
+    def update_by_name(
+        self,
+        name: str,
+        prompt: str | None = None,
+        config: str | None = None,
+        scope: str | None = None,
+        type: str | None = None,
+        valid_classes: str | None = None,
+    ):
+        with Session(self._engine) as session:
+            statement = select(Tag).where(Tag.name == name)
+            result = session.exec(statement).first()
+
+            if result:
+                # Update the status and date_updated fields
+                if prompt is not None:
+                    result.prompt = prompt
+
+                if config is not None:
+                    result.config = config
+
+                if type is not None:
+                    result.type = type
+
+                if scope is not None:
+                    result.scope = scope
+
+                if valid_classes is not None:
+                    if result.type != TagType.classification.value:
+                        result.meta = {}
+                    else:
+                        result.meta = {"valid_classes": valid_classes}

```

Comment:
using an `update` statement imo could eliminate those exhaustive checks for None